internal/contributebot/design/merge.md

```diff
@@ -0,0 +1,98 @@
+# Contribute Bot Auto-Merge Design
+
+Author: [@zombiezen][]<br>
+Date: 2018-12-17<br>
+Part of [#687][]
+
+[@zombiezen]: https://github.com/zombiezen
+[#687]: https://github.com/google/go-cloud/issues/687
+
+## Objective
+
+As a maintainer of Go Cloud, it is desirable to have pull requests merged in as
+quickly as they can be merged after receiving approvals, assuming there are no
+test breakages or merge conflicts. Currently, merging requires a manual process:
+
+```
+while branch is not up-to-date with master {
+  press update branch button
+  wait for Travis run to finish
+}
+click squash-and-merge
+```
+
+This loop is especially expensive the longer the waiting is, since it increases
+the likelihood that the branch is not up-to-date and thus causing the cycle to
+increase iterations (more waiting!).
+
+The objective of this design is to reduce the amount of maintainer [toil][] in
+this process.
+
+[toil]: https://landing.google.com/sre/sre-book/chapters/eliminating-toil/
+
+## Overview
+
+To reduce maintainer toil, this document proposes a new set of interactions for
+Contribute Bot. If a user writes a comment on a pull request with the text
+"/merge", Contribute Bot will:
+
+1.  Verify that the commenter has `write` or `admin` access to the repository.
+    If they do not, do not proceed and leave an error comment on the PR.
+2.  Verify that the pull request has the necessary approvals. If it does not, do
```

Comment:
Which necessary approvals? In particular, Github requires only a single approver, but our convention is to wait for all assignees to approve.

Reply:
The GitHub branch protections (added link and changed prose to clarify). I am assuming that members of the team with write access will use this appropriately.